### DISCUSSION TITLE
Missing tensor

### DISCUSSION THREAD

**Discussion by FloGro3** [explosion/spaCy#10802] [Help: Coding & Implementations] [answered]
As stated in the documentation: _"We also calculate an alignment between the word-piece tokens and the spaCy tokenization, so that we can use the last hidden states to set the Doc.tensor attribute."_ [Spacy Docu](https://spacy.io/api/transformer )

Here comes my problem (and I don't know if this is an issue or if I'm missing something):
![image](https://user-images.githubusercontent.com/93530403/168345452-bee1dcdb-b0a9-4e24-b37d-18c09ae561c5.png)

Here's the code to reproduce the problem:

```
import spacy  
spacy.require_gpu()
nlp = spacy.load("de_dep_news_trf")
doc = nlp("Dies ist ein Test.")
doc.tensor 
```

As you can see I use the "de_dep_news_trf" Pipeline from Spacy, but the tensor is empty and I don't know why. 

Glad for any feedback :)

**Comment by koaning** [comment, marked as answer]
You are likely interested in the [transformer attributes](https://spacy.io/api/transformer#assigned-attributes), which can be found via; 

```
Doc._.trf_data
```

This gives you access to a bunch of [attributes](https://spacy.io/api/transformer#transformerdata), it sounds like you'll be interested in the tensors, via; 

```
Doc._.trf_data.tensors
```

**Reply by koaning** [reply]
You might also appreciate [this resource](https://applied-language-technology.mooc.fi/html/notebooks/part_iii/05_embeddings_continued.html#contextual-word-embeddings-from-transformers) as you're exploring these contextualised embeddings.
